utils/src/main/scala/za/co/absa/enceladus/utils/broadcast/BroadcastUtils.scala

```diff
@@ -125,70 +104,68 @@ object BroadcastUtils {
-      case n => throw new IllegalArgumentException(s"Mapping UDFs with $n arguments are not supported. Should be between 1 and 10.")
-    }
-  }
-
-  private def getMultipleMappingLambda(mappingTable: Broadcast[LocalMappingTable],
-                                       defaultValues: Map[String, Any], numberOfArguments: Int): AnyRef = {
-    val defaultRow = if (defaultValues.isEmpty) {
+  private def getMappingLambda(mappingTable: Broadcast[LocalMappingTable],
+                               defaultValues: Map[String, Any], numberOfArguments: Int, isSingleOutput: Boolean): UserDefinedFunction = {
+    val defaultValue = if (isSingleOutput) {
+      defaultValues.get(mappingTable.value.outputColumns.values.head).orNull
+    } else if (defaultValues.isEmpty) {
       null
     } else {
       Row.fromSeq(mappingTable.value.outputColumns.values.toSeq.map(field => defaultValues.getOrElse(field, null)))
     }
+    val valueType: DataType = if(isSingleOutput) mappingTable.value.valueTypes.head else {
+      val structFields: Seq[StructField] = mappingTable.value.outputColumns.keys
+        .map(outputName => if (outputName.contains(".")) outputName.split("\\.").last else outputName).toSeq
+        .zip(mappingTable.value.valueTypes)
+        .map { case (name: String, fieldType: DataType) => StructField(name, fieldType) }
+      StructType(structFields)
+    }
     numberOfArguments match {
-      case 1 => (p1: Any) => mappingTable.value.getRowWithDefault(Seq(p1), defaultRow)
-      case 2 => (p1: Any, p2: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2), defaultRow)
-      case 3 => (p1: Any, p2: Any, p3: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2, p3), defaultRow)
-      case 4 => (p1: Any, p2: Any, p3: Any, p4: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4), defaultRow)
-      case 5 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any) => mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5), defaultRow)
-      case 6 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6), defaultRow)
-      case 7 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7), defaultRow)
-      case 8 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8), defaultRow)
-      case 9 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any, p9: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8, p9), defaultRow)
-      case 10 => (p1: Any, p2: Any, p3: Any, p4: Any, p5: Any, p6: Any, p7: Any, p8: Any, p9: Any, p10: Any) =>
-        mappingTable.value.getRowWithDefault(Seq(p1, p2, p3, p4, p5, p6, p7, p8, p9, p10), defaultRow)
+      case 1 => udf(new UDF1[Any, Any] {
```

Comment:
User Defined Aggregate Functions are not relevant here. 
`udf((p1: Any) => mappingTable.value.getRowWithDefault(Seq(p1), defaultValue))` will fail with `Schema for type Any is not supported`, since Spark 3.x changed the behavior, so the only option left is to use Java UDFs. I can explain more in detail if you want why this is the only option left